application/src/main/java/org/togetherjava/tjbot/commands/help/UserBannedDeleteRecentThreadsListener.java

```diff
@@ -0,0 +1,59 @@
+package org.togetherjava.tjbot.commands.help;
+
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.events.guild.GuildBanEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.requests.RestAction;
+import org.jooq.Record1;
+import org.jooq.Result;
+
+import org.togetherjava.tjbot.commands.EventReceiver;
+import org.togetherjava.tjbot.db.Database;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+
+import static org.togetherjava.tjbot.db.generated.Tables.HELP_THREADS;
+
+/**
+ * Reacts on users being banned from a guild. It then deletes (not just closes) all their recently
+ * created help threads.
+ * <p>
+ * This for example tackles threads created by scammers, which are otherwise not automatically
+ * cleaned up on ban.
+ */
+public final class UserBannedDeleteRecentThreadsListener extends ListenerAdapter
+        implements EventReceiver {
+    private static final Duration RECENT_THREAD_DURATION = Duration.ofMinutes(30);
+    private final Database database;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param database to find help threads created by the banned user
+     */
+    public UserBannedDeleteRecentThreadsListener(Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public void onGuildBan(GuildBanEvent event) {
+        RestAction.allOf(getRecentHelpThreads(event.getUser()).stream()
+            .map(record -> event.getJDA().getThreadChannelById(record.value1()))
+            .filter(Objects::nonNull)
+            .map(ThreadChannel::delete)
+            .toList()).queue();
+    }
+
+    private Result<Record1<Long>> getRecentHelpThreads(User user) {
```

Comment:
i think it would be better if this would return `List<Long>` instead. easier to understand and u essentially just unwrap it into that on the other side anyways.

Reply:
is this ok?

Reply:
done